dist_tables/ddl.rst

```diff
@@ -61,6 +61,30 @@ Each created shard is assigned a unique shard id and all its replicas have the s
 
 You are now ready to insert data into the distributed table and run queries on it. You can also learn more about the UDF used in this section in the :ref:`user_defined_functions` of our documentation.
 
+.. _colocation_groups:
+
+Co-Location Groups
+------------------
+
+Table rows are stored in shards across a Citus cluster. When all rows needed by a query are available in a single shard then the tables named by the query are said to be *co-located for the query*. (Tables which are co-located for the majority of queries in an application are simply called co-located tables.) Queries with co-located tables excute faster because Citus doesn't need to shuffle data over the network.
```

Comment:
This paragraph is slightly inaccurate. Looking around for some good definitions, I saw we have a good blog post on this topic: https://www.citusdata.com/blog/2016/12/22/scaling_out_sql_with-colocation/. 

These paragraphs are good intro ones, and provide a succinct definition:
> Co-location is the practice of dividing data tactically, where one keeps related information on the same machines to enable efficient relational operations, but takes advantage of the horizontal scalability for the whole dataset.

> The principle of data co-location is that all tables in the database have a common distribution column and are sharded across machines in the same way such that rows with the same distribution column value are always on the same machine, even across different tables. As long as the distribution column provides a meaningful grouping of data, relational operations can be performed within the groups.

The section in the blog: `Data co-location in Citus for hash-distributed tables`, up until (and including) the diagram is all good content to add to documentation.